hack/tools/tilt-prepare/main.go

```diff
@@ -70,10 +73,25 @@ var (
 	toolsFlag            = pflag.StringSlice("tools", []string{}, "list of tools to be created; each value should correspond to a make target")
 	certManagerFlag      = pflag.Bool("cert-manager", false, "prepare cert-manager")
 	kustomizeBuildsFlag  = pflag.StringSlice("kustomize-builds", []string{}, "list of kustomize build to be run; each value should be in the form name:path")
-	providersBuildsFlag  = pflag.StringSlice("providers", []string{}, "list of providers to be installed; each value should be in the form name:path[:debug]")
+	providersBuildsFlag  = pflag.StringSlice("providers", []string{}, "list of providers to be installed; each value should be in the form name:path")
 	allowK8SContextsFlag = pflag.StringSlice("allow-k8s-contexts", []string{}, "Specifies that Tilt is allowed to run against the specified k8s context name; Kind is automatically allowed")
+	tiltSettingsFileFlag = pflag.String("tilt-settings-file", "", "Path to a tilt-settings.(json|yaml) file")
 )
 
+type tiltSettings struct {
+	Debug     map[string]debugConfig `json:"debug,omitempty"`
+	ExtraArgs map[string]extraArgs   `json:"extra_args,omitempty"`
+}
```

Comment:
I'm fine with this as a first iteration, but let's open an issue about reading more information from the tilt file and dropping corresponding flags:
- allow-k8s-contexts
- cert-manager
- kustomize_build_arg 
- providers

Reply:
Sure, I will open an issue for this.

Reply:
See: https://github.com/kubernetes-sigs/cluster-api/issues/6246

Reply:
I wonder if we should additionally either hard-code tools, or set `kustomize,envsubst` as default value. 

I'm not sure it has value that this is configurable from the outside. We always need kustomize and envsubst when we want to deploy a provider (which is always).

If we assume that other folks might want to build additional tools, we can keep the flag and just set a default value.

Reply:
Just an idea for another follow-up. Now that we're going into the direction of having a fully specified config file, should we add an example tilt settings file which we can refer to in the documentation?

I always found it very useful for golangci-lint: https://github.com/golangci/golangci-lint/blob/master/.golangci.example.yml

Looking at our tilt documentation we now have a lot of configuration fields, and I think a well-documented example could be very useful.

(I can open a PR and take my local tilt-settings + documentation comments as basis)

WDYT?

Reply:
Yes, it looks like `tools_arg` is hard-coded to `kustomize,envsubst` in the Tiltfile, as it is not read anywhere there via e.g. `settings.get`. The only way to set this is if you call tilt-prepare directly.

Reply:
> Just an idea for another follow-up. Now that we're going into the direction of having a fully specified config file, should we add an example tilt settings file which we can refer to in the documentation?
> 
> I always found it very useful for golangci-lint: https://github.com/golangci/golangci-lint/blob/master/.golangci.example.yml
> 
> Looking at our tilt documentation we now have a lot of configuration fields, and I think a well-documented example could be very useful.
> 
> (I can open a PR and take my local tilt-settings + documentation comments as basis)
> 
> WDYT?

Sounds good :+1: